### DISCUSSION TITLE
Can I send commands to tasmota devices?

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16034] [Q&A] [answered]
I understand that each device has a built in web server but is there a way to send commands to different devices all on the same wifi network?
The linux device would just be another client on the same wireless lan network as all of the tasmoto devices would be.
I basically just want a way to centralize controlling all of the devices.

**Comment by sfromis** [comment, marked as answer]
Yes, the primary control method is using MQTT, and is build on sending commands that way
https://tasmota.github.io/docs/MQTT/#commands-over-mqtt

While you *can* also use http for commands, this method does not provide full featured Tasmota.

**Reply by dac2020** [reply]
MQTT, got it. I'll start looking around to see if I can learn more about it. A quick look shows I can get an MQTT client on openwrt, raspberry and others so running a 'server' or centralizing should mean simply a client that can be used as a single point to send/receive for all devices.

Just found out about tasmota a day or two ago so just learning and appreciate the help :).

**Reply by dac2020** [reply]
I'm seeing something called MQTT brokers while looking for info about building a MQTT server.
Are there open source servers/hosts I could self host?

**Reply by joba-1** [reply]
some say servers to mqtt brokers and mean the same. I use (self host) mosquitto, works fine.

**Reply by sfromis** [reply]
While the term broker is commonly used, you could also call it a "server" except that it only facilitates communication between devices. As a starting point, I can also recommend the Mosquitto broker. Open source, low footprint, and easy to install/operate. It's what I'm using, and it "just works". Any MQTT client can publish messages to, or subscribe from other clients, allowing for easy automation.

**Reply by dac2020** [reply]
Is the MQTT broker an open source project? I'm reading something about a MQTT Broker and pro Pro Edition for Eclipse Mosquitto.

It's not clear if the difference is mainly just getting support or not being able to use the free version for commercial purposes for example. It's also not clear if Eclipse is some sort of commercial venture outside of the MQTT broker project. 

Lots and lots fo figure out.

**Reply by sfromis** [reply]
There is no single "MQTT broker project", but Mosquitto is a good well-known [open-source](https://github.com/eclipse/mosquitto) broker. As the [project home page](https://mosquitto.org/) states:

> Eclipse Mosquitto is an open source (EPL/EDL licensed) message broker that implements the MQTT protocol versions 5.0, 3.1.1 and 3.1. Mosquitto is lightweight and is suitable for use on all devices from low power single board computers to full servers.

A quick search did not bring anything up about a "pro edition" Mosquitto. Only references for "pro edition" were other products able to work with Mosquitto. Mosquitto is managed as part of the Eclipse "family", but that does not detract from the full open source status. Obviously, if you need "professional support" with a service level agreement, you many need to find someone taking money for such a service, but that's not really related to the project status as open source.

And lots of people are actively using Mosquitto without worrying about support. As it is pretty stable, support would mostly be for "protection" against screwing up yourself, or perhaps for "more advanced" setups than basic unproblematic usage.

In short: Don't worry, be Happy!

**Reply by dac2020** [reply]
Pro edition is mentioned here.
http://www.steves-internet-guide.com/install-mosquitto-linux/

So, I think you are telling me that anyone can use this in any way including commercial and the only difference is that the pro version comes with support.

**Reply by sfromis** [reply]
> Pro edition is mentioned here.

Specifically what exact words are you referring to? Quote?

Neither of words "edition", "commercial" nor "pro" are used anywhere on that page (which seems to be focused on installation and basic usage), and "version" is not using in any context about support or commercial usage. And "support" only in the context of features available with a new version from a long time ago.

According to Wikipedia, the [EPL](https://en.wikipedia.org/wiki/Eclipse_Public_License) used for Mosquitto is specifically listed as "free software" by the FSF and "approved" by OSI. While the licence is meant to be "business-friendly", one restriction is that distributed modifications are required to grant recipients licence to relevant patents covering modifications.

**Reply by dac2020** [reply]
![2022-07-22_102544](https://user-images.githubusercontent.com/9727841/180492617-db563d3d-4359-4a4c-acee-186ba8dcd2f5.jpg)

**Reply by dac2020** [reply]
@sfromis I think the question has been answered at this point. Thank you for the additional information which helps to clarify.
Like I said, a lot to learn and just started a few days ago :)

**Reply by sfromis** [reply]
So it was not on the page as such, just an ad with "questionable wording". Going to the home page of that company, the pitch that they sell hosting, management interface, and support for running Mosquitto "high availability", which in no way affects other usage of the free open source broker.

**Reply by dac2020** [reply]
It's on the same page and since it's all new to me, there is no 'questionable wording'. 

Anyhow, no idea why you're so focused on this, you already made your point and I get it now so it's done.

**Reply by sfromis** [reply]
It is very much "questionable wording" to lead people to think that it is about a "pro edition" of the broker, when it is not. Not a big deal, that's how ads are. As a seasoned Internet user I automatically separate the published article from ads on the side, to earn income.

**Reply by dac2020** [reply]
The ad is on the same site as the installation notes so it's easy to think that it's related.
Anyhow, no big deal, I get it and appreciate the info you gave me :). All new to me so give me a while to get up to speed.
